tests/unit/backend/layers/api/test_portal_api.py

```diff
@@ -458,8 +458,10 @@ def test__post_collection_ignores_metadata_if_crossref_exception(self):
         )
         self.assertEqual(201, response.status_code)
         collection_id = json.loads(response.data)["collection_id"]
-        collection = self.business_logic.get_collection_version(CollectionVersionId(collection_id))
-        self.assertIsNone(collection.publisher_metadata)
+        # TODO: this endpoint should also return `version_id`
+        collection = self.business_logic.get_collection_version_from_canonical(CollectionId(collection_id))
+        if collection is not None:
+            self.assertIsNone(collection.publisher_metadata)
```

Comment:
When is the if logic tested?

Reply:
the `if` is required otherwise pylance will complain since `collection` can be an optional`. Unfortunately adding an assertion `.isNotNone` won't help, so you will find some of those ifs in tests.

Reply:
Are there flags like `#noqa` for pylance to ignore a line. Otherwise, this makes our tests more confusing by adding a forked path in the testing logic that may or may not be hit.